plugins/ltac/g_rewrite.mlg

```diff
@@ -276,10 +276,14 @@ VERNAC COMMAND EXTEND AddSetoid1 CLASSIFIED AS SIDEFF
      {
          add_setoid atts binders a aeq t n
      }
-  | #[ atts = rewrite_attributes; ] ![ maybe_open_proof ] [ "Add" "Morphism" constr(m) ":" ident(n) ]
-    (* This command may or may not open a goal *)
-    => { (if Lib.is_modtype() then VtSideff([n]) else VtStartProof(GuaranteesOpacity, [n])), VtLater }
-    -> { add_morphism_infer atts m n }
+  | #[ atts = rewrite_attributes; ] ![ open_proof ] [ "Add" "Morphism" constr(m) ":" ident(n) ]
+    => { VtStartProof(GuaranteesOpacity, [n]), VtLater }
+    -> { if Lib.is_modtype () then 
+           CErrors.user_err Pp.(str "Add Morphism cannot be used in a module type. Use Parameter Morphism instead.");
```

Comment:
I don't think we should forbid using it in module types.

Reply:
The old code was given to the same syntax 2 completely different behaviors depending if it was part of a module type of not. Hence the new syntax above. The error is there so that I can see failures in CI, we should probably put a warning to be nice to users ": please use XXX instead" and support for one version of Coq both.